tests/test_app/TestApp/Controller/AuthTestController.php

```diff
@@ -41,7 +42,7 @@ public function __construct($request = null, $response = null)
         parent::__construct($request, $response);
     }
 
-    public function initialize()
+    public function initialize(): void
```

Comment:
Is updating the various hook methods in an application going to add value or frustration? I left return typehints off of the hook methods in `Shell` and `Command` as I didn't see them adding much value to a developer's life, but they certainly are a big bump when upgrading.

Reply:
I also see your argument.
But: Using automated sniffers/fixers (existing), it is rather easy to fix up project level or plugins at once. 

The benefit of having a consistent code base is a requirement for this, though. If only some methods had typehinting the sniffer would fix too much and then you run into fatal errors as signature changes.

Also: The benefit for hooks is especially important (since they cannot be IDE completed/validated). People might try to return a response object or other weird things. So having a clear communicated API for them seems very useful.

Reply:
> The benefit of having a consistent code base is a requirement for this, though. If only some methods had typehinting the sniffer would fix too much and then you run into fatal errors as signature to parent is not compatible.

Sub-classes can add additional type restrictions that the parent class doesn't have in general.

> People might try to return a response object or other weird things. So having a clear communicated API for them seems very useful.

This could be a valuable point of having more types on hooks. It makes the framework's expectations of userland code clear. My concern is primarily around the initial upgrade and how bumpy that is going to be.

Reply:
Signature changes should be regex fixable etc.